source/PlayerInfo.cpp

```diff
@@ -1346,6 +1346,20 @@ void PlayerInfo::ReorderShip(int fromIndex, int toIndex)
 	shared_ptr<Ship> ship = ships[fromIndex];
 	ships.erase(ships.begin() + fromIndex);
 	ships.insert(ships.begin() + toIndex, ship);
+	// Check if the ship in the first position can be a flagship and is in the current system.
+	if(Preferences::Has("Automatically unpark flagship")
+			&& ships[0]->CanBeFlagship() && ships[0]->GetSystem() == this->GetSystem())
```

Comment:
if you have access to the private variable returned by a getter function, use the variable
```suggestion
			&& ships[0]->CanBeFlagship() && ships[0]->GetSystem() == system)
```